From dc8c7f5901b09bd65c806154ee1d7973cf449a25 Mon Sep 17 00:00:00 2001
From: Dmitry Shifrin <dmitry.shifrin@cogentembedded.com>
Date: Thu, 8 Sep 2016 20:46:37 +0300
Subject: [PATCH 1/2] Check width align

We have to check cam->out_width because it is a requirement of VnIS register
---
 drivers/media/platform/soc_camera/rcar_vin.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/soc_camera/rcar_vin.c b/drivers/media/platform/soc_camera/rcar_vin.c
index 92e1102..7fb3235 100644
--- a/drivers/media/platform/soc_camera/rcar_vin.c
+++ b/drivers/media/platform/soc_camera/rcar_vin.c
@@ -1663,10 +1663,6 @@ static int rcar_vin_set_fmt(struct soc_camera_device *icd,
 	if (mf.code != xlate->code)
 		return -EINVAL;
 
-	/* Prepare VIN crop */
-	cam->width = mf.width;
-	cam->height = mf.height;
-
 	/* Use VIN scaling to scale to the requested user window. */
 
 	/* We cannot scale up */
@@ -1683,6 +1679,14 @@ static int rcar_vin_set_fmt(struct soc_camera_device *icd,
 		pix->height = vin_sub_height;
 	}
 
+	if (can_scale && !IS_ALIGNED(pix->width, 0x10)){
+		return -EINVAL;
+	}
+
+	/* Prepare VIN crop */
+	cam->width = mf.width;
+	cam->height = mf.height;
+
 	/*
 	 * We have calculated CFLCR, the actual configuration will be performed
 	 * in rcar_vin_set_bus_param()
-- 
1.9.1

